mne/channels/montage.py

```diff
@@ -1682,6 +1744,101 @@ def read_polhemus_fastscan(fname, unit='mm'):
     return points
 
 
+def _read_eeglab_locations(fname, unit):
+    # VALID_SCALES = dict(mm=1e-3, cm=1e-2, m=1)
+    # _scale = _check_unit_and_get_scaling(unit, VALID_SCALES)
+
+    ch_names = np.genfromtxt(fname, dtype=str, usecols=3).tolist()
+    topo = np.loadtxt(fname, dtype=float, usecols=[1, 2])
+    sph = _topo_to_sph(topo)
+    pos = _sph_to_cart(sph)
+    pos[:, [0, 1]] = pos[:, [1, 0]] * [-1, 1]
+
+    return ch_names, pos
+
+
+def read_standard_montage(fname, head_size=HEAD_SIZE_DEFAULT, unit='m'):
+    """Read a standard montage file containing polar coordinates.
+
+    Parameters
+    ----------
+    fname : str
+        File extension is expected to be '.loc', '.locs' or '.eloc'.
+    head_size : float | None
+        The size of the head in [m]. If none, returns the values read from the
+        file with no modification.
+        Defaults to HEAD_SIZE_DEFAULT.
+
+    Returns
+    -------
+    montage : instance of DigMontage
+        The montage.
+
+    See Also
+    --------
+    make_dig_montage
+    make_standard_montage
+    """
+    from itertools import chain
+    from ._standard_montage_utils import (
+        _read_theta_phi_in_degrees, _read_sfp, _read_csd, _read_elc,
+        _read_elp_besa, _read_brainvision
+    )
+    SUPPORTED_FILE_EXT = {
+        'eeglab': ('.loc', '.locs', '.eloc', ),
+        'hydrocel': ('.sfp', ),
+        'matlab': ('.csd', ),
+        'asa electrode': ('.elc', ),
+        'generic (Theta-phi in degrees)': ('.txt', ),
+        'legacy mne-c': ('.hpts', ),
+        'standard BESA spherical': ('.elp', ),  # XXX: not same as polhemus elp
+        'brainvision': ('.bvef', ),
+    }
+
+    _, ext = op.splitext(fname)
+    _check_option('fname', ext, list(chain(*SUPPORTED_FILE_EXT.values())))
+
+    if ext in SUPPORTED_FILE_EXT['eeglab']:
+        if head_size is None:
+            raise(ValueError,
+                  "``head_size`` cannot be None for '{}'".format(ext))
+        ch_names, pos = _read_eeglab_locations(fname, unit)
+        scale = head_size / np.median(np.linalg.norm(pos, axis=-1))
+        pos *= scale
+
+        montage = make_dig_montage(
+            ch_pos=dict(zip(ch_names, pos)),
+            coord_frame='head',
+        )
+
+    elif ext in SUPPORTED_FILE_EXT['hydrocel']:
+        montage = _read_sfp(fname, head_size=head_size)
+
+    elif ext in SUPPORTED_FILE_EXT['matlab']:
+        montage = _read_csd(fname, head_size=head_size)
+
+    elif ext in SUPPORTED_FILE_EXT['asa electrode']:
+        montage = _read_elc(fname, head_size=head_size)
+
+    elif ext in SUPPORTED_FILE_EXT['generic (Theta-phi in degrees)']:
+        if head_size is None:
+            raise(ValueError,
+                  "``head_size`` cannot be None for '{}'".format(ext))
+        montage = _read_theta_phi_in_degrees(fname, head_size=head_size)
+
+    elif ext in SUPPORTED_FILE_EXT['legacy mne-c']:
+        montage = read_dig_polhemus_fastscan(fname, unit=unit)
```

Comment:
shall we remove read_dig_polhemus_fastscan from public API? typically this can contain
digitized location. It's not necessarily in polar. Also it's not a polhemus format. It's historical format required by mne-c code. See https://www.nmr.mgh.harvard.edu/mne/stable/manual/c_reference.html?highlight=hpts#mne-convert-dig-data